models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/AnthropicStreamFunctionCallingHelper.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 - 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.ai.anthropic.api;
+
+import org.springframework.ai.anthropic.api.AnthropicApi.MediaContent;
+import org.springframework.ai.anthropic.api.AnthropicApi.StreamResponse;
+import org.springframework.ai.model.ModelOptionsUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper class to support streaming function calling.
+ * <p>
+ * It can merge the streamed {@link StreamResponse} chunks in case of function calling
+ * message.
+ *
+ * @author Mariusz Bernacki
+ * @since 1.0.0
+ */
+public class AnthropicStreamFunctionCallingHelper {
+
+	public boolean isStreamingToolFunctionCall(StreamResponse response) {
+		if (response == null || response.contentBlock() == null) {
+			return false;
+		}
+
+		return response.contentBlock().type() == MediaContent.Type.TOOL_USE;
+	}
+
+	public boolean isStreamingToolFunctionCallFinish(StreamResponse response) {
```

Comment:
Not sure this is correct. 
Isn't it possible to have multiple "tool_use" content blocks in the same message? Perhaps for parallel function calling? 
According to the [streaming spec](https://docs.anthropic.com/en/api/messages-streaming#streaming-request-with-tool-use) a message can include multiple content blocks including tools such. 

Maybe we should use the content block stop event instead?